relay-server/src/metrics_extraction/transactions.rs

```diff
@@ -1,20 +1,473 @@
+            name,
+            unit,
+        };
+
+        Metric::new(mri, value, timestamp, tags)
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+struct TransactionsMeasurementTags {
+    measurement_rating: Option<String>,
+    release: Option<String>,
+    dist: Option<String>,
+    environment: Option<String>,
+    transaction: Option<String>,
+    platform: Option<String>,
+    transaction_status: Option<String>,
+    transaction_op: Option<String>,
+    http_method: Option<String>,
+    other: BTreeMap<String, String>,
+}
+
+impl TransactionsMeasurementTags {
+    fn new(measurement_rating: Option<String>, universal_tags: UniversalTags) -> Self {
+        Self {
+            measurement_rating,
+            release: universal_tags.release,
+            dist: universal_tags.dist,
+            environment: universal_tags.environment,
+            transaction: universal_tags.transaction,
+            platform: universal_tags.platform,
+            transaction_status: universal_tags.transaction_status,
+            transaction_op: universal_tags.transaction_op,
+            http_method: universal_tags.http_method,
+            other: universal_tags.other,
+        }
+    }
+}
+
+impl From<TransactionsMeasurementTags> for BTreeMap<String, String> {
+    fn from(value: TransactionsMeasurementTags) -> Self {
+        let mut map = value.other;
+
+        if let Some(measurement_rating) = value.measurement_rating {
+            map.insert("measurement_rating".to_string(), measurement_rating);
+        }
+
+        if let Some(release) = value.release {
+            map.insert("release".to_string(), release);
+        }
+
+        if let Some(dist) = value.dist {
+            map.insert("dist".to_string(), dist);
+        }
+
+        if let Some(environment) = value.environment {
+            map.insert("environment".to_string(), environment);
+        }
+
+        if let Some(transaction) = value.transaction {
+            map.insert("transaction".to_string(), transaction);
+        }
+        if let Some(platform) = value.platform {
+            map.insert("platform".to_string(), platform);
+        }
+        if let Some(transaction_status) = value.transaction_status {
+            map.insert("transaction.status".to_string(), transaction_status);
+        }
+        if let Some(transaction_op) = value.transaction_op {
+            map.insert("transaction.op".to_string(), transaction_op);
+        }
+        if let Some(http_method) = value.http_method {
+            map.insert("http.method".to_string(), http_method);
+        }
+
+        map
+    }
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+struct TransactionsBreakdownTags {
+    release: Option<String>,
+    dist: Option<String>,
+    environment: Option<String>,
+    transaction: Option<String>,
+    platform: Option<String>,
+    transaction_status: Option<String>,
+    transaction_op: Option<String>,
+    http_method: Option<String>,
+    other: BTreeMap<String, String>,
```

Comment:
I did not check line-by-line, but is this type identical to `TransactionMeasurementTags`? If so, they can be one type.